packages/core/content-manager/server/src/services/permission-checker.ts

```diff
@@ -12,76 +10,81 @@ const ACTIONS = {
       return userAbility.cannot(action, toSubject(entity), field);
     };
 
-    const sanitizeOutput = (data, { action = ACTIONS.read } = {}) => {
+    const sanitizeOutput = (data: any, { action = ACTIONS.read } = {}) => {
       return permissionsManager.sanitizeOutput(data, { subject: toSubject(data), action });
     };
 
-    const sanitizeQuery = (query, { action = ACTIONS.read } = {}) => {
+    const sanitizeQuery = (query: any, { action = ACTIONS.read } = {}) => {
       return permissionsManager.sanitizeQuery(query, { subject: model, action });
     };
 
-    const sanitizeInput = (action, data, entity) => {
+    const sanitizeInput = (action: any, data: any, entity?: any) => {
       return permissionsManager.sanitizeInput(data, {
         subject: entity ? toSubject(entity) : model,
         action,
       });
     };
 
-    const validateQuery = (query, { action = ACTIONS.read } = {}) => {
+    const validateQuery = (query: any, { action = ACTIONS.read } = {}) => {
       return permissionsManager.validateQuery(query, { subject: model, action });
     };
 
-    const validateInput = (action, data, entity) => {
+    const validateInput = (action: any, data: any, entity: any) => {
       return permissionsManager.validateInput(data, {
         subject: entity ? toSubject(entity) : model,
         action,
       });
     };
 
-    const sanitizeCreateInput = (data) => sanitizeInput(ACTIONS.create, data);
-    const sanitizeUpdateInput = (entity) => (data) => sanitizeInput(ACTIONS.update, data, entity);
+    const sanitizeCreateInput = (data: any) => sanitizeInput(ACTIONS.create, data);
+    const sanitizeUpdateInput = (entity: any) => (data: any) =>
+      sanitizeInput(ACTIONS.update, data, entity);
 
-    const buildPermissionQuery = (query, action) => {
+    const buildPermissionQuery = (query: any, action: any) => {
       return permissionsManager.addPermissionsQueryTo(query, action);
     };
 
     /**
      * @param {string} query
      * @param {keyof typeof ACTIONS} action
      */
-    const sanitizedQuery = (query, action) => {
+    const sanitizedQuery = (query: any, action: any) => {
       return pipeAsync(
         (q) => sanitizeQuery(q, action),
         (q) => buildPermissionQuery(q, action)
       )(query);
     };
 
     // Sanitized queries shortcuts
-    Object.keys(ACTIONS).forEach((action) => {
-      sanitizedQuery[action] = (query) => sanitizedQuery(query, ACTIONS[action]);
+    Object.keys(ACTIONS).forEach((action: any) => {
+      // @ts-expect-error TODO
+      sanitizedQuery[action] = (query: any) => sanitizedQuery(query, ACTIONS[action]);
     });
 
     // Permission utils shortcuts
     Object.keys(ACTIONS).forEach((action) => {
-      can[action] = (...args) => can(ACTIONS[action], ...args);
-      cannot[action] = (...args) => cannot(ACTIONS[action], ...args);
+      // @ts-expect-error TODO
+      can[action] = (...args: any) => can(ACTIONS[action], ...args);
```

Comment:
@Marc-Roig this confuses me a bit

the ts error is `type 'string' can't be used to index type '(action: any, entity: any, field: any) => any` which makes sense to me as the only can defined here is the func at L23. The same problem seems to exist before this TS change though

Reply:
maybe this refers to the "action" which is maybe not interpreted as a string, can you try to define the ACTIONS object as :

```ts
const ACTIONS = {
  read: 'plugin::content-manager.explorer.read',
  create: 'plugin::content-manager.explorer.create',
  update: 'plugin::content-manager.explorer.update',
  ...
} as const;
```

Reply:
ah 💡 missunderstood what was happening. Yes, I dont know if this can actually be properly typed 🤔  maybe @Convly  knows more about this.

These are shortcuts aadded on top of what originally is a function, javascript allows this sourcery and probably typescript has some magic too. Lets leave it as a ts epxect error for now if it's not easy to fix